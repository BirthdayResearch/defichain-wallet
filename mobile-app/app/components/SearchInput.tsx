import { ThemedIcon, ThemedTextInput, ThemedView } from '@components/themed'
import { ClearButton } from '@components/WalletTextInput'
import { tailwind } from '@tailwind'

import { TextInputProps } from 'react-native'

type SearchInputProps = React.PropsWithChildren<TextInputProps> & ISearchInputProps

interface ISearchInputProps {
  showClearButton: boolean
  onClearInput: () => void
}

export function SearchInput (props: SearchInputProps): JSX.Element {
  const {
    onClearInput,
    ...otherProps
  } = props
  return (
    <ThemedView
      light={tailwind('bg-gray-100')}
      dark={tailwind('bg-gray-900')}
      style={tailwind('rounded-lg flex flex-row items-center py-1 pl-2 flex-1')}
    >
      <ThemedIcon
        iconType='MaterialIcons'
        name='search'
        size={16}
        light={tailwind('text-gray-600')}
        dark={tailwind('text-gray-300')}
        style={tailwind('mr-2')}
      />
      <ThemedTextInput
        {...otherProps}
        style={tailwind('flex-grow w-8/12 h-8')}
      />
      {props.showClearButton &&
        (
          <ClearButton
            onPress={onClearInput}
            iconThemedProps={{
              light: tailwind('text-gray-300'),
              dark: tailwind('text-gray-600')
            }}
          />
        )}
    </ThemedView>
  )
}
