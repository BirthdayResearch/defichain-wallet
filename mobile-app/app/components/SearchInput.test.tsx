import { render } from '@testing-library/react-native'
import { SearchInput } from './SearchInput'

jest.mock('@shared-contexts/ThemeProvider')

describe('Search Input', () => {
  it('should match snapshot', () => {
    const rendered = render(
      <SearchInput
        showClearButton
        onClearInput={jest.fn()}
      />)
    expect(rendered.toJSON()).toMatchSnapshot()
  })
})
