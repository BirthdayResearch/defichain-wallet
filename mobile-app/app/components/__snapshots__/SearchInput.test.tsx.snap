// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Input should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
        "display": "flex",
        "flexBasis": "0%",
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 4,
        "paddingLeft": 8,
        "paddingTop": 4,
      },
      Object {
        "backgroundColor": "rgba(245, 245, 245, 1)",
      },
    ]
  }
>
  <Text />
  <TextInput
    allowFontScaling={true}
    placeholderTextColor="rgba(0, 0, 0, 0.4)"
    rejectResponderTermination={true}
    showClearButton={true}
    style={
      Array [
        Object {
          "flexGrow": 1,
          "height": 32,
          "width": "66.666667%",
        },
        Object {
          "color": "rgba(64, 64, 64, 1)",
        },
      ]
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "opacity": 1,
        "paddingBottom": 2,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 2,
        "position": "relative",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "borderBottomLeftRadius": 9999,
            "borderBottomRightRadius": 9999,
            "borderTopLeftRadius": 9999,
            "borderTopRightRadius": 9999,
            "height": 16,
            "left": 12,
            "position": "absolute",
            "top": 8,
            "width": "75%",
            "zIndex": -1,
          },
          Object {
            "backgroundColor": "rgba(38, 38, 38, 1)",
          },
        ]
      }
    />
    <Text />
  </View>
</View>
`;
